capstone/src/main/webapp/scripts/tags.js

```diff
@@ -33,3 +37,41 @@ function openAboutTags() {
   modal.style.display = "block"
   addModalListeners('about-tags-modal', 'about-close');
 }
+
+
+/** 
+ * Opens a modal that shows the usage of a tag in the context of 
+ * the group's posts, comments, etc. 
+ */
+function openTagsContext(tag) {
+  const tagNameElement = document.getElementById('tag-name');
+  tagNameElement.innerHTML = 'Usage of tag: ' + tag;
+
+  fetch(`/tags-context?groupId=${groupId}&tag=${tag}`)
+  .then(response => response.json())
+  .then((contextText) => {  
+    let modal = document.getElementById('tags-context-modal');
+    displayTagContext(contextText);
+    modal.style.display = "block"
+    addModalListeners('tags-context-modal', 'context-close');
+  });
+}
+
+/** Display the context of a given tag. */
+function displayTagContext(contextText) {
+  const container = document.getElementById('context-container');
+  container.innerHTML = "";
+
+  for (context of contextText) {
+    const textElement = document.createElement('p');
+    textElement.id = 'context-text';
+    const boldedText = boldTag(context.text, context.tagText);
+    textElement.innerHTML = `${context.type}:	${boldedText}`;
+    container.appendChild(textElement);
+  }
+}
+
+/** Bold the given part of a text. */
+function boldTag(text, tag) {
+  return text.replace(new RegExp('(^|\\W)(' + tag + ')(\\W|$)','ig'), '$1<strong>$2</strong>$3');
```

Comment:
I like this approach! Double checking, would this allow for multiple occurrences to be bolded in the same string? From what I can tell, you've marked this as case insensitive AND global with the 'ig', so I think the answer should be yes.

Reply:
Yes, it will!